### PR TITLE
feat(migrations): support monitoring.type to none

### PR DESCRIPTION
**Pull request by ralgozino**
- Add support for changing the monitoring type from prometheus to none and from mimir to none. This will effectively delete the monitoring stack from the cluster.
- Prometheus to Mimir, and Mimir to Prometheus are not supported.
- Add e2e test for Mimir to None.

scenario tested with an on-prem cluster:
- deployed with `montioring.type = prometheus`, verified monitoring stack working.
- changed to `monitoring.type = none`, verified removal of monitoring stack.
- changed to `monitoring.type = mimir`, verified monitoring stack working.
- changed to `monitoring.type = none`, verified removal of monitoring stack.

Fixes https://github.com/sighupio/product-management/issues/324

### CODE REVIEW

File: templates/distribution/scripts/pre-apply.sh.tpl [hunk 1]
```diff
@@ -324,6 +324,79 @@ deleteVeleroMinio
...
+
+deletePrometheusOperator() {
+  $kustomizebin build $vendorPath/modules/monitoring/katalog/prometheus-operator | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  echo "Prometheus Operator resources deleted"
+}
+
+deletePrometheusOperated() {
+  # we first delete the CRs before deleting the CRDs to avoid the `kubectl delete` command from failing due to unexisting APIs.
+  $kustomizebin build $vendorPath/modules/monitoring/katalog/prometheus-operated | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  echo "Waiting 3 minutes for PVCs to liberate"
```
Comment (nutellinoit) [inline templates/distribution/scripts/pre-apply.sh.tpl:360]: This wait is really needed?

File: templates/distribution/scripts/pre-apply.sh.tpl [hunk 2]
```diff
@@ -324,6 +324,79 @@ deleteVeleroMinio
...
+  sleep 180
+  $kubectlbin delete -l app.kubernetes.io/name=prometheus,app.kubernetes.io/instance=k8s pvc -n monitoring --wait --timeout=180s
+  echo "Prometheus Operated resources deleted"
+}
+
+deleteMimir() {
+  $kustomizebin build $vendorPath/modules/monitoring/katalog/mimir | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  $kustomizebin build $vendorPath/modules/monitoring/katalog/minio-ha | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  echo "Waiting 3 minutes for PVCs to liberate"
```
Comment (nutellinoit) [inline templates/distribution/scripts/pre-apply.sh.tpl:369]: Same as above

**Review by nutellinoit — CHANGES_REQUESTED** [review]
Aside from the 3 minutes sleep, LGTM

**Comment by nutellinoit** [comment]
I will merge it, let's do a double check on all the migrations this afternoon
